etl/steps/data/garden/ihme_gbd/2019/gbd_tools.py

```diff
@@ -152,12 +153,177 @@ def add_share_of_population(df: pd.DataFrame) -> pd.DataFrame:
+            "short_unit": "",
+            "num_decimal_places": 1,
+        },
+        "Incidence - Number": {
+            "title": f"Number of new cases of {cause.lower()}, in {sex.lower()} aged {age.lower()}",
+            "description": "",
+            "unit": "cases",
+            "short_unit": "",
+            "num_decimal_places": 0,
+        },
+        "Prevalence - Number": {
+            "title": f"Current number of cases of {cause.lower()}, in {sex.lower()} aged {age.lower()}",
+            "description": "",
+            "unit": "cases",
+            "short_unit": "",
+            "num_decimal_places": 0,
+        },
+        "Incidence - Rate": {
+            "title": f"Number of new cases of {cause.lower()} per 100,000 people, in {sex.lower()} aged {age.lower()}",
+            "description": "",
+            "unit": "cases",
+            "short_unit": "",
+            "num_decimal_places": 1,
+        },
+        "Prevalence - Rate": {
+            "title": f"Current number of cases of {cause.lower()} per 100,000 people, in {sex.lower()} aged {age.lower()}",
+            "description": "",
+            "unit": "cases",
+            "short_unit": "",
+            "num_decimal_places": 1,
+        },
+        "Incidence - Share of the population": {
+            "title": f"Number of new cases of {cause.lower()} per 100 people, in {sex.lower()} aged {age.lower()}",
+            "description": "",
+            "unit": "%",
+            "short_unit": "%",
+            "num_decimal_places": 1,
+        },
+        "Prevalence - Share of the population": {
+            "title": f"Current number of cases of {cause.lower()} per 100 people, in {sex.lower()} aged {age.lower()}",
+            "description": "",
+            "unit": "%",
+            "short_unit": "%",
+            "num_decimal_places": 1,
+        },
+    }
+    new_variable = variable.copy()
+    new_variable.name = underscore(var_name_dict[variable.name]["title"])
+    new_variable.metadata = VariableMeta(
+        title=var_name_dict[variable.name]["title"],
+        description=var_name_dict[variable.name]["description"],
+        unit=var_name_dict[variable.name]["unit"],
+        short_unit=var_name_dict[variable.name]["short_unit"],
+    )
+    new_variable.metadata.display = {
+        "name": var_name_dict[variable.name]["title"],
+        "numDecimalPlaces": var_name_dict[variable.name]["num_decimal_places"],
+    }
+
+    return new_variable
+
+
+def add_metadata_and_save(dest_dir: str, ds_meadow: Dataset, df: pd.DataFrame, dims: List[str]) -> None:
+    """
+    Adding metadata at the variable level
+    """
+    ds_garden = Dataset.create_empty(dest_dir)
+    ds_garden.metadata = ds_meadow.metadata
+
+    df = df.reset_index()
+    df_group = df.groupby(dims)
+    for group_id, group in df_group:
+        dims_id = dict(zip(dims, group_id))
+        cause = dims_id["cause"]
+        sex = dims_id["sex"]
+        age = dims_id["age"]
```

Comment:
Maybe you don't need to define these variables? I see you only use them in line 292, where you could just use the dictionary directly.

Actually, since the keys in the dictionary are named the same as the arguments from `create_variable_metadata`, you could use the following for line 292:

```
cleaned_variable = create_variable_metadata(variable=tb_group[variable_name], **dims_id)
```

Reply:
Oh wow, that's new python wizardry for me!